core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java

```diff
@@ -14,243 +14,457 @@
-      public int compare(DistributionAggregation o1, DistributionAggregation o2) {
-        return Long.valueOf(o1.getCount()).compareTo(o2.getCount());
-      }
-    });
-
-    DistributionAggregation distributionAggregation1 = distributionAggregations.get(0);
-    DistributionAggregation distributionAggregation2 = distributionAggregations.get(1);
-
-    verifyDistributionAggregation(distributionAggregation1, 1, 10.0, 10.0, 10.0, 10.0, 1);
-    verifyDistributionAggregation(distributionAggregation2, 2, 80.0, 40.0, 30.0, 50.0, 1);
-    verifyBucketCounts(distributionAggregation1.getBucketCounts(), 9);
-    verifyBucketCounts(distributionAggregation2.getBucketCounts(), 14, 16);
-    assertThat(distributionAggregation1.getTags().get(0).getKey()).isEqualTo(tagKey);
-    assertThat(distributionAggregation1.getTags().get(0).getValue()).isEqualTo(tagValue1);
-    assertThat(distributionAggregation2.getTags().get(0).getKey()).isEqualTo(tagKey);
-    assertThat(distributionAggregation2.getTags().get(0).getValue()).isEqualTo(tagValue2);
+    statsManager.registerView(
+        DistributionViewDescriptor.create(
+            VIEW_NAME,
+            VIEW_DESCRIPTION,
+            MEASUREMENT_DESCRIPTOR,
+            DISTRIBUTION_AGGREGATION_DESCRIPTOR,
+            Arrays.asList(KEY)));
+    statsManager.record(
+        createContext(factory, KEY, VALUE), MeasurementMap.of(MEASUREMENT_DESCRIPTOR, 10.0));
+    statsManager.record(
+        createContext(factory, KEY, VALUE2), MeasurementMap.of(MEASUREMENT_DESCRIPTOR, 30.0));
+    statsManager.record(
+        createContext(factory, KEY, VALUE2), MeasurementMap.of(MEASUREMENT_DESCRIPTOR, 50.0));
+    DistributionView view = (DistributionView) statsManager.getView(VIEW_NAME);
+    assertDistributionAggregationsEquivalent(
+        TOLERANCE,
+        view.getDistributionAggregations(),
+        Arrays.asList(
+            StatsTestUtil.createDistributionAggregation(
+                Arrays.asList(Tag.create(KEY, VALUE)), BUCKET_BOUNDARIES, Arrays.asList(10.0)),
+            StatsTestUtil.createDistributionAggregation(
+                Arrays.asList(Tag.create(KEY, VALUE2)),
+                BUCKET_BOUNDARIES,
+                Arrays.asList(30.0, 50.0))));
   }
 
-  private static void verifyDistributionAggregation(
-      DistributionAggregation distributionAggregation,
-      int count, double sum, double mean, double min, double max, int tagsSize) {
-    assertThat(distributionAggregation.getCount()).isEqualTo(count);
-    assertThat(distributionAggregation.getSum()).isWithin(TOLERANCE).of(sum);
-    assertThat(distributionAggregation.getMean()).isWithin(TOLERANCE).of(mean);
-    assertThat(distributionAggregation.getRange().getMin()).isWithin(TOLERANCE).of(min);
-    assertThat(distributionAggregation.getRange().getMax()).isWithin(TOLERANCE).of(max);
-    assertThat(distributionAggregation.getTags().size()).isEqualTo(tagsSize);
+  @Test
+  public void allowRecordingWithoutRegisteringMatchingView() {
```

Comment:
I left this test, even though it no longer throws an exception, because I thought it would be useful to check that calling `record` without registering a view doesn't throw an exception.  There is nothing to assert, because there are no views to check.  I added a comment to explain the purpose of the test.